models.py

```diff
@@ -138,13 +138,61 @@ def get_items(item_list, controller):
                 return_list.append(item)
         return return_list
 
+    def split_items(self):
```

Comment:
I'm not sure my idea is the best one, but maybe this could be a dynamic `property` with the same name? Usage example:
`switches, unis, links = self.split_items`

Or maybe we could do this at write-time: `items` would be a `property` that, when updated, would populate `self.switches`, `unis` and `links`.